python/ALU_Control.py

```diff
@@ -1,11 +1,39 @@
 from myhdl import *
 
 @block
-def ALU_Control(clk, op, out): #Clk and OpCode input, int output
+def ALU_Control(ALUop, funct, ALUcontrol): 
 
-    @always(clk.posedge)
-    def execute():
-        #Insert op code interpretation
-        return
+	@always(ALUop, funct)
+	def control():
+		if (ALUop == 0):
+			ALUcontrol = 0
 
-    return execute
+		elif (ALUop == 1):
+			ALUcontrol = 1
+
+		elif (ALUop == 2):
+			ALUcontrol = 0
+
+		elif (ALUop == 3):
+			ALUcontrol = 1
+
+		elif (ALUop == 4):
+			if (funct == 0):
+				ALUcontrol = 0
+
+			elif (funct == 2):
+				ALUcontrol = 1
+
+			elif (funct == 6):
+				ALUcontrol = 2
+
+return ALU_Control
+
+def JR_Control(alu_op, funct, JRControl):
```

Comment:
There needs to be an @always(...) or an @always_comb here, I would recommend @always_comb.